python/plugins/processing/tools/dataobjects.py

```diff
@@ -70,7 +70,11 @@ def createContext(feedback=None):
     context.setInvalidGeometryCheck(invalid_features_method)
 
     settings = QgsSettings()
-    context.setDefaultEncoding(settings.value("/Processing/encoding", "System"))
+    default_codec = "System"
+    system_codec = QTextCodec.codecForLocale()
+    if (system_codec):
+        default_codec = system_codec.name().data().decode()
+    context.setDefaultEncoding(settings.value("/Processing/encoding", default_codec))
```

Comment:
I'd flip this around a little --  currently the logic is reliant on the gui upgrading an existing "system" settings value to an actual codec name. But that assumes that the algorithm will always be run through the gui. 

So maybe something like this is more appropriate:

```suggestion
    encoding_name = settings.value("/Processing/encoding", "System")
    if not encoding_name or encoding_name == "System":
        system_codec = QTextCodec.codecForLocale()
        if system_codec:
          encoding_name = system_codec.name().data().decode()
    context.setDefaultEncoding(encoding_name)
```

Reply:
Or even better -- move this logic to QgsProcessingUtils (as "resolveDefaultEncoding") and call it from both here and QgsProcessingLayerOutputDestinationWidget::selectEncoding(). That'll avoid the duplicate logic.

Reply:
Done, thanks. I didn't want to add any functions since we were in bugfix / freeze, but this is indeed the most elegant solution. :)